src/main/java/atreides/task/TaskList.java

```diff
@@ -0,0 +1,148 @@
+        return newTask;
+    }
+
+    public TaskList(ArrayList<String[]> tasks) {
+        this.list = new ArrayList<>();
+        for (String[] words : tasks) {
+            Task newTask = new Task();
+            Boolean isDone = words[1].equals("1");
+            if (words[0].equals("T")) {
+                newTask = new ToDo(words[2]);
+            } else if (words[0].equals("D")) {;
+                String description = words[2];
+                String by = words[3];
+                newTask = new Deadline(description, by);
+            } else if (words[0].equals("E")) {
+                String description = words[2];
+                String[] startEnd = words[3].split("-");
+                newTask = new Events(description, startEnd);
+            }
+            newTask.markDone(isDone);
+            this.list.add(newTask);
+        }
+    }
+
+    public TaskList() {
+        this.list = new ArrayList<>();
+    }
+
+    /**
+     * Creates string of list of tasks that the Ui will print out
+     * @return
+     */
+    public String showList() {
+        StringBuilder tasks = new StringBuilder();
+        for (int i = 0; i < list.size(); i++) {
+            tasks.append((i + 1)).append(".").append(list.get(i));
+            if (i != list.size() - 1) {
+                tasks.append("\n");
+            }
+        }
+        return tasks.toString();
+    }
+
+    /**
+     * creates a string of list of tasks that will be written into the file object
+     * @return
+     */
+    public String writeList() {
+        StringBuilder tasks = new StringBuilder();
+        for (int i = 0; i < list.size(); i++) {
+            tasks.append(list.get(i).write());
+            if (i != list.size() - 1) {
+                tasks.append("\n");
+            }
+        }
+        return tasks.toString();
+    }
+
+    public void mark(int index) throws AtreidesException{
+        markBool(index, true);
+    }
+
+    public void unmark(int index) throws AtreidesException{
+        markBool(index, false);
+    }
+
+    /**
+     * marks whether task at the specific index based on the bool
+     * @param index
+     * @param bool
+     * @throws AtreidesException If index out of bounds
+     */
+    public void markBool(int index, boolean bool) throws AtreidesException{
+        checkIndexPresent(index);
+        list.get(index).markDone(bool);
+    }
+
+    public void checkIndexPresent(int index) throws AtreidesException {
+        if (index >= list.size()) {
+            throw new AtreidesException("list does not have the index present");
+        }
+    }
+
+    public Task getTask(int index) {
+        return list.get(index);
+    }
+
+    public int size() {
+        return list.size();
+    }
+
+    public Task delete(int index) throws AtreidesException {
+        checkIndexPresent(index);
+        return list.remove(index);
+    }
+
+    public void addTask(Task task) {
+        list.add(task);
+    }
+
+    public String find(String description) {
```

Comment:
Should this method be named findTask(s) or have a header to make it clear what it does?